### ISSUE TITLE
toml parser cuts off the first letter of the first key on the first line

### ISSUE THREAD

**Issue by Apache-HB** [marzer/tomlplusplus#58]
## Environment
**Compiler:**  

g++ 9.3 Ubuntu

**C++ standard mode (e.g. 17, 20, 'latest'):**  

C++17

**Target arch (e.g. x64):**  

x86_64 for wsl

**Library configuration overrides:**  

TOML_EXCEPTIONS=0

**Relevant compilation flags:**  

-fno-exceptions -fno-rtti

## Describe the bug

when a toml file has no newlines or spaces before the first key the first letter is cut off

## Steps to reproduce (or a small repro code sample)

test.toml
```toml
jeff = "name"
```

```cpp
auto it = toml::parse_file("test.toml");
auto tab = std::move(it).get();

auto jeff = tab["jeff"].value_or<std::string>("no"sv);
print(jeff); // prints "no" instead of "name"
```

## Additional information

im using WSL on windows 2004

**Comment by marzer** [comment]
Hmmmn, this is certainly covered by tests already, so definitely shouldn't be happening. Can you try replacing the call to `toml::parse_file` with `toml::parse(R"(jeff = "name")"sv)` and letting me know what happens?

**Comment by marzer** [comment]
I can't reproduce this in the current version. I just tried it on GCC9 (Ubuntu via WSL on windows 2004, same as you), and I see the correct result of `"name"`.

Given that you're using a deprecated function that should have been caught with a warning, my suggestion would be to update the library and try again. There was a bug relating to an invalid short-cirtuit in `value_or` that was fixed in [v2.1.0](https://github.com/marzer/tomlplusplus/releases/tag/v2.1.0); my suspicion is that's the bug you're seeing.
